scripts/lint-datagrid.test.sh

```
@@ -0,0 +1,346 @@
+  ((TESTS_PASSED++))
+  echo ""
+}
+
+fail_test() {
+  echo -e "${RED}✗ FAIL: $1${NC}"
+  ((TESTS_FAILED++))
+  echo ""
+}
+
+# Setup test environment
+cd "$TEST_DIR"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)